iosapp/lib-ios/Tests/LibAppTests/IOSReducerTests.swift

```diff
@@ -0,0 +1,483 @@
+import Combine
+import ComposableArchitecture
+import GertieIOS
+import LibClients
+import LibCore
+import XCTest
+import XExpect
+
+@testable import LibApp
+
+enum SavedCodable: Equatable {
+  case protectionMode(ProtectionMode)
+  case disabledBlockGroups([BlockGroup])
+}
+
+final class IOSReducerTests: XCTestCase {
+  func testHappyPath() async throws {
```

Comment:
this test is a bit of a beast, but it really shows of the upsides of TCA. i'm able to take the whole app through a very long series of events and assert on every state change, and every side-effect/dependency that i care about is controlled and asserted on.